pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -195,6 +195,25 @@ func TestExecTracer(t *testing.T) {
 				require.Equal(t, events[0].Cwd, cwd, "Event has bad cwd")
 			},
 		},
+		"event_failed": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+					GetFailed:  true,
+				}
+			},
+			generateEvent: func() (int, error) {
+				cmd := exec.Command("/bin/foobar")
+				cmd.Run()
+
+				return 0, nil
+			},
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, _ int, events []types.Event) {
+				fmt.Printf("events: %v\n", events)
+				require.Len(t, events, 1, "One event expected")
+				require.Equal(t, []string{"/bin/foobar"}, events[0].Args, "Event has bad args")
```

Comment:
Shouldn't we test that the return value was actually set? For instance:

```go
				require.NotEqual(t, int(0), events[0].Retval, "Event has bad retval")
```

Or we could directly verify it is `-2` as we know that's the error we are triggering. I'm fine with both solutions, maybe checking `0` is more generic.